File: crates/enforcer-memory/tests/fixtures/memory/lang_ruby/widget.rb
require 'json'
require_relative './helper'

class Animal
  def speak
    puts "..."
  end
end

class Widget < Animal
  def initialize(name)
    @name = name
  end

  def draw
    if @name.empty?
      return "unnamed"
    end
    helper(@name)
  end

  def self.create
    Widget.new("default")
  end
end

def helper(label)
  label.upcase
end

File: crates/enforcer-memory/tests/fixtures/memory/lang_lua/widget.lua
local json = require("json")

local Widget = {}

function Widget.new(name)
  local self = { name = name }
  return self
end

function Widget:draw()
  if self.name == "" then
    return "unnamed"
  end
  return helper(self.name)
end

function helper(label)
  return json.encode(label)
end

local w = Widget.new("world")
w:draw()
